### ISSUE TITLE
Create indexes for preferred bibles automatically 

### ISSUE THREAD

**Issue by agrogers** [AndBible/and-bible#1869]
When we enable preferred bibles, automatically create indexes for those translations.

**Comment by tuomas2** [comment]
I don't really like this idea because it is such a slow process to build indices. First startup would take too long. And not everyone even need indices (for every module). 

If we want user friendliness, better to improve descriptions etc.

**Comment by tuomas2** [comment]
Myself I am using index only in 1 or 2 modules, and for the rest I don't care.

**Comment by agrogers** [comment]
I would agree if the process has to be done in the foreground. But you have
got the cool background creation which is what I almost always use. So I
don't think it needs to slow anything down.

And I also only use it on a few translations which is why I think it should
be done only on preferred bibles, not all documents.

I raise the issue because I feel like I have created and waited for indexes
half a dozen times over the last week or so. But I haven't had that
experience on any of the other Bible apps I have been testing out.

**Comment by tuomas2** [comment]
Well maybe maybe... In fact I always wait until index is created so I was not even thinking about the background task.

Perhaps there should be a "Index everything in the background" function as a first step, and then we could initiate that (optionally?) on the first run.

**Comment by tuomas2** [comment]
Should also make sure that indexing here can't crash anything. I don't want that this causes first-time users to experience crashes...

**Comment by agrogers** [comment]
I have never had a problem with background indexing.

It would be good to have a progress bar somewhere showing the indexing.
Just a thin line at the bottom of the action bar might work. But if the
indexing happened by default in the background very few people would even
notice it and search would just work.

**Comment by tuomas2** [comment]
In app notification there is progress bar?

**Comment by agrogers** [comment]
Oh, is there. I don't look there 🙄.
